### PR TITLE
convert full node settings to functional components

### PR DESCRIPTION
**Pull request by benma**
Converted some components to React functional components. One (electrum-server.jsx) is outstanding. Keeping for later to keep the PR a bit smaller.

### CODE REVIEW

File: frontends/web/src/routes/settings/electrum-servers.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      <div className="row">
+        <div className={['flex flex-row flex-between flex-items-center', style.titleContainer].join(' ')}>
+          <h4 className="subTitle m-none">{t('settings.electrum.servers')}</h4>
+          <A href="#" className={['labelLarge labelLink', style.resetLink].join(' ')} onClick={resetToDefault}>{t('settings.electrum.reset')}</A>
+        </div>
+        <ul className={style.servers}>
+          {
+            electrumServers.map((server, index) => (
+              <ElectrumServer
+                // @ts-ignore
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-servers.tsx:97]: Hm, does this do anything? I can't see any warnings on my side without it.

File: frontends/web/src/routes/settings/electrum.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2018 Shift Devices AG
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum.tsx:2]:
```suggestion
 * Copyright 2022 Shift Devices AG
```

Perhaps you forgot this one :)

File: frontends/web/src/routes/settings/electrum-servers.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+type Props = {
+  coin: 'btc' | 'tbtc' | 'ltc' | 'tltc';
+};
+
+interface IElectrumServer {
+  server: string;
+  tls: boolean;
+  pemCert: string;
+}
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-servers.tsx:33]:
```suggestion
type IElectrumServer = {
  server: string;
  tls: boolean;
  pemCert: string;
}
```
As we are trying to move away from interfaces to types, we can convert this one.

File: frontends/web/src/routes/settings/electrum-servers.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  const save = async (newElectrumServers: IElectrumServer[]) => {
+    const currentConfig = await getConfig();
+    currentConfig.backend[coin].electrumServers = newElectrumServers;
+    await setConfig(currentConfig);
+    setConfigState(currentConfig);
+  };
+
+  const onAdd = (server: IElectrumServer) => {
+    let newElectrumServers = electrumServers.slice();
+    newElectrumServers.push(server);
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-servers.tsx:58]: Unsure if this is a part of the PR, but might as well refactor this while we're at it.

```suggestion
    const newElectrumServers = [...electrumServers, server];
```

File: frontends/web/src/routes/settings/electrum-servers.tsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    save(newElectrumServers);
+  };
+
+  const onRemove = (index: number) => {
+    let newElectrumServers = electrumServers.slice();
```
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-servers.tsx:63]: same like the `onAdd`, `electrumServers` can be shallowly copied using the spread syntax:

```suggestion
    const newElectrumServers = [...electrumServers];
```
[slice vs spread perf](https://measurethat.net/Benchmarks/Show/2667/0/arrayprototypeslice-vs-spread-operator#latest_results_block)  - performance wise spread is good.

but prob not that it matters as much (browsers, machines, etc play a role in the perf), especially theres not a big difference in it. For me it's more so about following modern JS.

**Review by shonsirsha — CHANGES_REQUESTED** [review]
Looks good. Functionalities to: add, remove, reset, and test full node servers are working fine when I tested it manually! 👍 

Just some nitpicks :)

File: frontends/web/src/routes/settings/electrum-servers.tsx (hunk 5 above)
Comment (shonsirsha) [inline frontends/web/src/routes/settings/electrum-servers.tsx:63]: same like the `onAdd`, `electrumServers` can be shallowly copied using the spread syntax:

```suggestion
    const newElectrumServers = [...electrumServers];
```

File: frontends/web/src/routes/settings/electrum-servers.tsx (hunk 1 above)
Comment (benma) [reply frontends/web/src/routes/settings/electrum-servers.tsx]: Removed

File: frontends/web/src/routes/settings/electrum.tsx (hunk 2 above)
Comment (benma) [reply frontends/web/src/routes/settings/electrum.tsx:2]: Added a new copyright line

File: frontends/web/src/routes/settings/electrum-servers.tsx (hunk 4 above)
Comment (benma) [reply frontends/web/src/routes/settings/electrum-servers.tsx]: Neat! Done

**Review by shonsirsha — APPROVED** [review]
LGTM! 🙂

**Pull request merged**